includes/class-wc-payments-account.php

```diff
@@ -1342,6 +1343,20 @@ public function maybe_handle_onboarding() {
 				// in the "everything OK" scenario).
 				if ( WC_Payments_Onboarding_Service::is_test_mode_enabled() ) {
 					try {
+						$account = $this->get_cached_account_data();
```

Comment:
I think we can move all this new logic to a method like `save_test_drive_settings`

Reply:
changed in [7f19936](https://github.com/Automattic/woocommerce-payments/pull/9928/commits/7f19936fdea8700071260772a7106e3e6f6ad51e) . thx